07_Virtual_environments_to_iterate_and_test/README.md

```diff
@@ -0,0 +1,48 @@
+# Extending containers with virtual environments for faster testing
+
+
+This is a short example of how to extend the containers built via `cotainr` via virtual environments. This approach can be useful for developing and testing as it doesn't require rebuilding a container from scratch every time a new package is added.
+
+> [!WARNING]
+> This should not be the default way of installing python packages as it puts a lot of strain on the Lustre file system. Once you have a complete set of python packages and their versions, always create a new container.
+
+## Requirements
+
+We assume you have built a container from a `conda` environment file via something like:
+```bash
+module load LUMI/23.03 cotainr
+cotainr build minimal_pytorch.sif --base-image=/appl/local/containers/sif-images/lumi-rocm-rocm-5.6.1.sif --conda-env=minimal_pytorch.yml
+```
+
+## Set up virtual environment
+
+First we run a shell inside the container
+```bash
+singularity shell --bind /pfs,/scratch,/projappl,/project,/flash,/appl minimal_pytorch.sif
+```
+Note that setting `--bind` is optional, you achieve the same by
+```bash
+module use /project/project_465001063/modules
+module load singularity-userfilesystems
+singularity shell minimal_pytorch.sif
+```
+
+In order to install additional packages we create a virtual environment via `venv` and activate it inside the container
+```bash
+python -m venv myenv --system-site-packages
+source myenv/bin/activate
+```
+The `--system-site-packages` flag gives the virtual environment access to the packages from the container.
+
+## Install custom packages
+
+After activating the virtual environment we can now install custom packages via pip, for example:
+```bash
+pip install torchmetrics
+```
+
```

Comment:
Maybe add a section showing how to correctly run the container with the venv, i.e. sourcing the venv.